test/controllers/visitors_controller_test.rb

```diff
@@ -42,4 +44,31 @@ class VisitorsControllerTest < ActionDispatch::IntegrationTest
     assert_redirected_to root_url
   end
 
+  test "when you are logged in as a regular user home page has only non-hidden courses" do
+    # sign in here
+    @user = users(:tim)
+    sign_in @user
+ 
+    stub_course_organization("course-org-1")
+    stub_course_organization("course-org-2")
+    hidden_course = Course.create!({ name: "hidden-course-1", course_organization: "course-org-1", hidden: true })
+    visible_course = Course.create!({ name: "visible-course-2", course_organization: "course-org-2", hidden: false })
+ 
+    visible_course_count = Course.where(hidden: false).count
```

Comment:
Because we're not adding a migration to prevent `hidden : null`, I'd personally add a case of that as well as flipping the logic to mirror the index method's `Course.where.not(hidden: true)`. It'd be **even better** if we could expose said filter method so we could consistently call it from the controller and test, but mirroring it is fine for now.

If you wanted to break that into a separate test that's fine too but at the bare minimum we should have a test exposing the `hidden : null` potential from the database.

Reply:
That's fair.    I'll add in a test for that.